pkg/registry/registry.go

```diff
@@ -471,7 +471,7 @@ func (r *PlainRegistry) listArchsWithAuth(ctx context.Context, client http.Clien
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to get manifest list. Unexpected status code %d. Expecting %d", resp.StatusCode, http.StatusOK)
+		return nil, fmt.Errorf("failed to get manifest list for image %s in registry %s. Unexpected status code %d. Expecting %d", image, registry, resp.StatusCode, http.StatusOK)
```

Comment:
It would be interesting to also know which authorization provider is being used.
It's a bit more work, but higher value